packages/eslint-plugin/src/rules/no-misused-object-likes.ts

```diff
@@ -0,0 +1,149 @@
+import type { Identifier, MemberExpression } from '@typescript-eslint/ast-spec';
+import type { TSESTree } from '@typescript-eslint/utils';
+
+import { createRule, getParserServices } from '../util';
+
+export type Options = [
+  {
+    checkObjectKeysForMap?: boolean;
+    checkObjectValuesForMap?: boolean;
+    checkObjectEntriesForMap?: boolean;
+    checkObjectKeysForSet?: boolean;
+    checkObjectValuesForSet?: boolean;
+    checkObjectEntriesForSet?: boolean;
+  },
+];
```

Comment:
[Question] Why allow such a granular set of options? Is there a known use case where someone would want to skip, say, only `Object.values`?